packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart

```diff
@@ -42,6 +44,34 @@ enum WKAudiovisualMediaType {
   all,
 }
 
+/// An object that contains information about an action that causes navigation to occur.
+///
+/// Wraps [WKNavigationAction](https://developer.apple.com/documentation/webkit/wknavigationaction?language=objc).
+class WKNavigationAction {
+  /// Constructs a [WKNavigationAction].
+  WKNavigationAction({required this.request, required this.targetFrame});
+
+  /// The URL request object associated with the navigation action.
+  final NSUrlRequest request;
+
+  /// The frame in which to display the new content.
+  final WKFrameInfo targetFrame;
+}
+
+/// An object that contains information about a frame on a webpage.
+///
+/// An instance of this class is a transient, data-only object; it does not
+/// uniquely identify a frame across multiple delegate method calls.
+///
+/// Wraps [WKFrameInfo](https://developer.apple.com/documentation/webkit/wkframeinfo?language=objc).
+class WKFrameInfo {
+  /// Construct a [WKFrameInfo].
+  WKFrameInfo({required this.isMainFrame});
+
+  /// Indicates whether the frame is the web site's main frame or a subframe.
+  final bool isMainFrame;
+}
+
 /// A script that the web view injects into a webpage.
 ///
 /// Wraps [WKUserScript](https://developer.apple.com/documentation/webkit/wkuserscript?language=objc).
```

Comment:
Could you also fix this in the two classes just below?

I've been hitting this a lot in doing analysis option updates, and it's hard to add later if it ends up public (which shouldn't happen for these, but better safe than sorry. E.g., in case we do decide to someday vend this layer as a plugin.)